### PR TITLE
Updated rmw_* return codes

### PR DESCRIPTION
**Pull request by ahcorde**
Updated return codes for unimplemented rmw_*_*_allocation functions. Related with this PR https://github.com/ros2/rmw_implementation/pull/137

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

**Review by hidmic — APPROVED** [review]
LGTM, but before merging, would you mind opening a PR against `rmw` to add this error code to the API spec?

**Comment by hidmic** [comment]
It's defined in `rmw`, but the API spec of the these functions does not mention that they can return such code (even though in practice they are unsupported).

File: rmw_connext_shared_cpp/src/wait_set.cpp [hunk 1]
```diff
@@ -142,7 +142,7 @@ destroy_wait_set(const char * implementation_identifier, rmw_wait_set_t * wait_s
   RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
     wait_set handle,
     wait_set->implementation_identifier, implementation_identifier,
-    return RMW_RET_ERROR)
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION)
```
Comment (hidmic) [inline rmw_connext_shared_cpp/src/wait_set.cpp:145]: @ahcorde shouldn't the one above be `RMW_RET_INVALID_ARGUMENT`? Also, consider using the  `RMW_CHECK_ARGUMENT_FOR_NULL` macro.

File: rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/wait.hpp [hunk 2]
```diff
@@ -36,39 +36,40 @@ __gather_event_conditions(
...
-      current_event->data)->get_entity();
-    if (!dds_entity) {
-      RMW_SET_ERROR_MSG("Event handle is null");
-      return RMW_RET_ERROR;
-    }
-    DDS::StatusCondition * status_condition = dds_entity->get_statuscondition();
-    if (!status_condition) {
-      RMW_SET_ERROR_MSG("status condition handle is null");
-      return RMW_RET_ERROR;
+  if (events) {
```
Comment (hidmic) [inline rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/wait.hpp:39]: @ahcorde why this change?

**Comment by hidmic** [comment]
I think the PR title needs clarification of what's being changed.

File: rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/wait.hpp (hunk 2 above)
Comment (ahcorde) [reply rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/wait.hpp:39]: Events might be NULL [as defined in the API](https://github.com/ros2/rmw/blob/145612d50ead587e156fbd8515417c44108de6d3/rmw/include/rmw/rmw.h#L2038). Without this change this function will return RMW_RET_ERROR.

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12472/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7439/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10185/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12392/)

**Review by hidmic — APPROVED** [review]

**Comment by hidmic** [comment]
CI up to `test_rmw_implementation` and `rcl`, against all Tier 1 RMW implementations:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12476/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7443/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10189/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12395/)
